src/solidity-frontend/solidity_convert.cpp

```diff
@@ -0,0 +1,1898 @@
+  {
+    // auxiliary type: pointer (FuncToPtr decay)
+    // This part is for FunctionToPointer decay only
+    assert(
+      type_name["typeString"].get<std::string>().find("function") !=
+      std::string::npos);
+
+    // Since Solidity does not have this, first make a pointee
+    nlohmann::json pointee = make_pointee_type(type_name);
+    typet sub_type;
+    if(get_func_decl_ref_type(pointee, sub_type))
+      return true;
+
+    if(
+      sub_type.is_struct() ||
+      sub_type.is_union()) // for "assert(sum > 100)", false || false
+      assert(!"struct or union is NOT supported");
+
+    new_type = gen_pointer_type(sub_type);
+    break;
+  }
+  case SolidityGrammar::TypeNameT::PointerArrayToPtr:
+  {
+    // auxiliary type: pointer (FuncToPtr decay)
+    // This part is for FunctionToPointer decay only
+    assert(
+      type_name["typeIdentifier"].get<std::string>().find("ArrayToPtr") !=
+      std::string::npos);
+
+    // Array type descriptor is like:
+    //  "typeIdentifier": "ArrayToPtr",
+    //  "typeString": "uint8[2] memory"
+
+    // Since Solidity does not have this, first make a pointee
+    typet sub_type;
+    if(
+      get_array_to_pointer_type(
+        type_name,
+        sub_type)) // TODO: Fix me! Inconsistency. Better to use Solidity compiler library?
+      return true;
+
+    if(
+      sub_type.is_struct() ||
+      sub_type.is_union()) // for "assert(sum > 100)", false || false
+      assert(!"struct or union is NOT supported");
+
+    new_type = gen_pointer_type(sub_type);
+    break;
+  }
+  case SolidityGrammar::TypeNameT::ArrayTypeName:
+  {
+    // Deal with array with constant size, e.g., int a[2]; Similar to clang::Type::ConstantArray
+    // array's typeDescription is in a compact form, e.g.:
+    //    "typeIdentifier": "t_array$_t_uint8_$2_storage_ptr",
+    //    "typeString": "uint8[2]"
+    // We need to extract the elementary type of array from the information provided above
+    // We want to make it like ["baseType"]["typeDescriptions"]
+    nlohmann::json array_elementary_type =
+      make_array_elementary_type(type_name);
+    typet the_type;
+    if(get_type_description(array_elementary_type, the_type))
+      return true;
+
+    assert(the_type.is_unsignedbv()); // assuming array size is unsigned bv
+    std::string the_size = get_array_size(type_name);
+    unsigned z_ext_value = std::stoul(the_size, nullptr);
+    new_type = array_typet(
+      the_type,
+      constant_exprt(
+        integer2binary(z_ext_value, bv_width(int_type())),
+        integer2string(z_ext_value),
+        int_type()));
+
+    break;
+  }
+  default:
+  {
+    msg.status(fmt::format(
+      "	@@@ got type name=SolidityGrammar::TypeNameT::{}",
+      SolidityGrammar::type_name_to_str(type)));
+    assert(!"Unimplemented type in rule type-name");
+    return true;
+  }
+  }
+
+  // TODO: More var decl attributes checks:
+  //    - Constant
+  //    - Volatile
+  //    - isRestrict
```

Comment:
Remember to open an issue

Reply:
ACK.
will do it when extending the new frontend to support more benchmarks.